lemur/plugins/lemur_cryptography/plugin.py

```diff
@@ -41,9 +41,10 @@ def issue_certificate(csr, options, private_key=None):
     if options.get("authority"):
         # Issue certificate signed by an existing lemur_certificates authority
         issuer_subject = options["authority"].authority_certificate.subject
-        assert (
-            private_key is None
-        ), "Private would be ignored, authority key used instead"
+        if "parent" not in options:
+            assert (
+                private_key is None
+            ), "Private would be ignored, authority key used instead"
```

Comment:
@raul-te, I am wondering if we shouldn't move this logic rather  to `def build_certificate_authority(options):`, since `private_key` would be anyways overwritten in case an authority is provided, and might make it more difficult to debug. Any thoughts?

for example
```
def build_certificate_authority(options):
    options["certificate_authority"] = True
    csr, private_key = create_csr(**options)
    
    if options.get("parent"):
        # Intermediate Cert Issuance
        cert_pem, chain_cert_pem = issue_certificate(csr, options, None)
    else:
        cert_pem, chain_cert_pem = issue_certificate(csr, options, private_key)

    return cert_pem, private_key, chain_cert_pem
```

Reply:
Yes, that makes sense. I've implemented your suggestion.

Reply:
awesome, thank you.